File: Python/Program/Motherboard.py
from KomponenInternal import KomponenInternal

class Motherboard(KomponenInternal):
    def __init__(self, merk="", nama="", harga=0, jenis_slot="", terpasang=False,
                 chipset="", form_factor=""):
        super().__init__(merk, nama, harga, jenis_slot, terpasang)
        self.chipset = chipset
        self.form_factor = form_factor
    
    # Setter
    def set_chipset(self, chipset):
        self.chipset = chipset
    
    def set_form_factor(self, form_factor):
        self.form_factor = form_factor
    
    # Getter
    def get_chipset(self):
        return self.chipset
    
    def get_form_factor(self):
        return self.form_factor
    
    # Tampilkan Info
    def tampilkan_info(self):
        return f"{super().tampilkan_info()} | Chipset: {self.chipset} | Form Factor: {self.form_factor}"

File: Java/Program/Motherboard.java
public class Motherboard extends KomponenInternal {
    private String chipset;
    private String formFactor;

    // Constructor
    public Motherboard() {
        super();
        this.chipset = "";
        this.formFactor = "";
    }

    public Motherboard(String merk, String nama, int harga, String jenisSlot, boolean terpasang,
                      String chipset, String formFactor) {
        super(merk, nama, harga, jenisSlot, terpasang);
        this.chipset = chipset;
        this.formFactor = formFactor;
    }

    // Setter methods
    public void setChipset(String chipset) { this.chipset = chipset; }
    public void setFormFactor(String formFactor) { this.formFactor = formFactor; }

    // Getter methods
    public String getChipset() { return chipset; }
    public String getFormFactor() { return formFactor; }

    // Tampilkan Info
    @Override
    public String tampilkanInfo() {
        return super.tampilkanInfo() + " | Chipset: " + chipset + " | Form Factor: " + formFactor;
    }
}